src/librustdoc/clean/types.rs

```diff
@@ -648,12 +647,38 @@ impl<'a> Iterator for ListAttributesIter<'a> {
 crate trait AttributesExt {
     /// Finds an attribute as List and returns the list of attributes nested inside.
     fn lists(&self, name: Symbol) -> ListAttributesIter<'_>;
+    
+    fn span(&self) -> Option<rustc_span::Span>;
+    
+    fn inner_docs(&self) -> bool;
+    
+    fn other_attrs(&self) -> Vec<ast::Attribute>;
 }
 
 impl AttributesExt for [ast::Attribute] {
     fn lists(&self, name: Symbol) -> ListAttributesIter<'_> {
         ListAttributesIter { attrs: self.iter(), current_list: Vec::new().into_iter(), name }
     }
+    
+    fn span(&self) -> Option<rustc_span::Span> {
+        self.iter()
+            .filter(|attr| attr.doc_str().is_some())
+            .map(|attr| attr.span)
+            .next()
+    }
+
+    fn inner_docs(&self) -> bool {
+        self.iter()
+        .find(|a| a.doc_str().is_some())
+        .map_or(true, |a| a.style == AttrStyle::Inner)
+    }
+
+    fn other_attrs(&self) -> Vec<ast::Attribute> {
+        self.iter().filter_map(|attr| match attr.doc_str() {
+            Some(_) => None,
+            None => Some(attr.clone()),
+        }).collect()
```

Comment:
```suggestion
    fn other_attrs(&self) -> impl Iterator<Item = &ast::Attribute> {
        self.iter().filter(|attr| attr.doc_str().is_none())
```

Reply:
Hmm, I realized while reviewing that `attr.doc_str().is_some()` and `attr.is_doc_comment()` behave differently :/ I'll look into fixing that.

Reply:
I noticed you change this to use `.filter()` but didn't change the return type - was that intentional? Did you run into trouble?

Reply:
The reason I suggested removing `.collect()` is it avoids an allocation for the common case where you just want to check each attribute one at a time.

Reply:
Ho ok. That's why !

I didn't notice the change on the return type, so yes I got some trouble and had to clone and collect the attributes. 
I will try your way instead !

Reply:
I tried, didn't go well ;)